### PR TITLE
Added a function that returns the intersection of two lists

### PR DESCRIPTION
**Pull request by Liya-F**

### CODE REVIEW

File: utilities/helper-functions.metta [hunk 1]
```diff
@@ -236,7 +236,30 @@
...
+(= (member $x $y)
+    (case $y
+        (
+            ( (:: $h $t) (if (== $x $h) True (member $x $t)))
+            ( () False)
+        )
+    )
+)
+
+; Function to return the intersection of two lists
```
Comment (Yagth) [inline utilities/helper-functions.metta:249]: It's also best if you test the function whether it works with the tree data structure I gave you yesterday and let me know.

Comment (Yagth) [reply utilities/helper-functions.metta:249]: I meant, the intersection function should work with list of tree nodes too

Comment (EyobAshenaki) [reply utilities/helper-functions.metta:249]: Isn't this implemented already? @Yagth

Comment (Yagth) [reply utilities/helper-functions.metta:249]: @EyobAshenaki Yeah I just saw it. But I don't think it works for a list of tree nodes. Does it?

Comment (EyobAshenaki) [reply utilities/helper-functions.metta:249]: Yes it doesn't. Maybe it can be modified to accommodate that instead of doing it from scratch.

**Pull request closed without merging**